### DISCUSSION TITLE
Migration from version 1 to 2

### DISCUSSION THREAD

**Discussion by ZlobnyiSerg** [elsa-workflows/elsa-core#464] [Q&A]
Hi. I see you actively working on version 2.0 which uses YesOrm for data persistence. Will it be possible to migrate from v1 to v2 without losing workflow data and without complex manual migrations?
We're going to use Elsa in production in month or two and we're in doubt which version to start from? Is it preview5 of version 2.0 is stable enough for production?

**Comment by sfmskywalker** [comment]
Hi @ZlobnyiSerg 

* Originally I wanted to use YesSQL as the ORM of choice and not abstract it away behind repository abstractions as to allow other modules to take advantage of the full querying capabilities, but I have a different approach in mind allowing the re-introduction of persistence abstractions, and as a result, the re-introduction of EF Core & MongoDB providers.

* The V2 workflow document schema has changed somewhat, but not significantly enough that it would be impossible to migrate workflow data.

* I *think* V2's core engine is stable enough to run in production, and a customer is going live with it as we speak, but we'll have to see once it's been up there for a while.

* V2 has no designer currently. If you need that, it's best to use V1. If you are using programmatic workflow only, I recommend going with the V2 previews (preview 6 to be released early next week which contains improvements and bug fixes).

**Comment by ZlobnyiSerg** [comment]
Thank you for response (I think part of these answers could become part of documentation for v2?).
Couple of other questions:
1. We are ok to design workflows manually, but not from code. Is it possible to design with v1 designer and manually adopt file contents to run on v2?
2. When you'll migrate back from YesSQL to EF Core, will DB schema stay the same? Will workflows running on YesSQL data structure be able to run on EF engine without impact/migrations?

**Comment by sfmskywalker** [comment]
1. Should be doable, although I haven't tried it.
2. No, YesSQL uses a different DB schema than which would be used with EF Core. If you were to switch providers from YesSQL to EF Core, the data would have to migrated. This could be done using code and leveraging both providers (YesSQL to read from and EF Core to write to).
